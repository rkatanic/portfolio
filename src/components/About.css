.about-item {
  background: var(--color-input);
  flex: 1;
  height: 100%;
  margin: 1rem 0;
  padding: 1rem 2rem;
  transition: 0.25s ease;
}

.about:hover .about-item {
  opacity: 0.5;
}

.about .about-item:hover {
  opacity: 1;
  transform: scale(1.1);
  cursor: default;
}

.about .about-item:hover .about-item-title::before {
  width: 3.5rem;
}

.about-item-title {
  text-transform: uppercase;
  letter-spacing: 2px;
  position: relative;
}

.about-item-title::before {
  position: absolute;
  content: "";
  bottom: 0;
  width: 2rem;
  height: 2px;
  background-color: var(--color-primary);
  transition: 0.5s ease;
}

.about-item-content {
  display: flex;
  flex-wrap: wrap;
  align-items: baseline;
  margin: 1rem 0;
  font-weight: 600;
}

.about-item-content-wrapper {
  flex: 1;
  min-width: 9rem;
}

.about-item-content-date {
  font-weight: 400;
  margin: 0 2rem 0 0;
  text-transform: uppercase;
  font-size: 0.75rem;
  color: var(--color-text);
  letter-spacing: 2px;
  min-width: 9rem;
}

.about-item-content-headline {
  margin: 0;
  color: var(--color-title);
  font-size: 1.25rem;
  letter-spacing: 1px;
}

.about-item-content-description {
  margin: 0;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: 500;
  font-size: 0.925rem;
  color: var(--color-primary);
}

.about-item-content-location {
  margin: 0;
  font-weight: 400;
  font-size: 0.75rem;
  text-transform: uppercase;
  letter-spacing: 1px;
}

#about .btn-outline {
  margin-top: 2rem;
  width: 100%;
  text-align: center;
}
