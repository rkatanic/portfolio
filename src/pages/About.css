.row {
  display: flex;
  align-items: center;
}

.number {
  letter-spacing: 2px;
  font-weight: 400;
  font-size: 0.825rem;
}

.content-text {
  font-weight: 400;
}
