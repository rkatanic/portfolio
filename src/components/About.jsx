import Section from "./Section";
import { ReactComponent as CalendarIcon } from "../assets/icons/calendar.svg";
import { ReactComponent as SuitcaseIcon } from "../assets/icons/suitcase.svg";
import { ReactComponent as EducationIcon } from "../assets/icons/education.svg";
import { ReactComponent as HtmlIcon } from "../assets/icons/html.svg";
import { ReactComponent as CssIcon } from "../assets/icons/css.svg";
import { ReactComponent as ReactIcon } from "../assets/icons/react.svg";
import { ReactComponent as AngularIcon } from "../assets/icons/angular.svg";
import myimage from "../assets/images/me2.png";

import "./About.css";

const About = () => {
  return (
    <Section id="about">
      <div className="about">
        <h1 className="about-title">About</h1>
        <h3 className="about-subtitle">My personal journey</h3>

        <div className="about-content">
          <div className="about-content-container">
            <h2 className="about-content-container-title">
              <SuitcaseIcon />
              Work
            </h2>

            <div style={{ position: "relative" }}>
              <div className="about-rounder"></div>
              <div className="about-line"></div>

              <h2>ProductDock</h2>
              <h3>Sotware developer</h3>
              <p>
                <CalendarIcon />
                August 2020 - present
              </p>
            </div>
            <div style={{ position: "relative" }}>
              <div className="about-rounder"></div>
              <h2>ProductDock Rookie Bootcamp</h2>
              <h3>Intern</h3>
              <p>
                <CalendarIcon />
                March 2020 - June 2020
              </p>
            </div>
          </div>

          <div className="about-content-container">
            <h2 className="about-content-container-title">
              <EducationIcon />
              Education
            </h2>

            <div style={{ position: "relative" }}>
              <div className="about-rounder"></div>
              <div className="about-line"></div>

              <h2>Slobomir P University</h2>
              <h3>IT Engineer</h3>
              <p>
                <CalendarIcon />
                2015 - 2019
              </p>
            </div>

            <div style={{ position: "relative" }}>
              <div className="about-rounder"></div>
              <h2>Electrotechnical High School in Doboj</h2>
              <h3>Computer technician</h3>
              <p>
                <CalendarIcon /> 2011 - 2015
              </p>
            </div>
          </div>
        </div>

        <h2 className="about-title">Skills</h2>
        <h3 className="about-subtitle">My technical orientation</h3>

        <div className="skill-content">
          <div className="skill-container">
            <HtmlIcon />
            <span>HTML</span>
          </div>
          <div className="skill-container">
            <CssIcon />
            <span>CSS</span>
          </div>
          <div className="skill-container">
            <ReactIcon />
            <span>React</span>
          </div>
          <div className="skill-container">
            <AngularIcon />
            <span>Angular</span>
          </div>
        </div>

        <div className="download-cv-container">
          <div className="download-cv-container-left">
            <h2 className="download-cv-container-title">
              Want more information?
            </h2>
            <p>
              For more details, feel free to download my CV and take a look!
            </p>
            <button className="btn btn-outline">Download CV</button>
          </div>
          <img src={myimage} alt="" />
        </div>
      </div>
    </Section>
  );
};

export default About;
