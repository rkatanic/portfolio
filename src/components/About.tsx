import Section from "./Section";
import Skills from "./Skills";

import "./About.css";
import Button from "./Button";

const About = (): JSX.Element => (
  <Section id="about">
    <h3>My personal journey</h3>
    <div className="about">
      <div className="about-item">
        <h3 className="about-item-title">Experience</h3>
        <div className="about-item-content">
          <p className="about-item-content-date">08/2019 - present</p>
          <div>
            <p className="about-item-content-headline">Software Developer</p>
            <p className="about-item-content-description">ProductDock</p>
          </div>
        </div>
        <div className="about-item-content">
          <p className="about-item-content-date">03/2019 - 06/2019</p>
          <div>
            <p className="about-item-content-headline">Intern</p>
            <p className="about-item-content-description">
              ProductDock Rookie Boot Camp
            </p>
          </div>
        </div>
      </div>
      <div className="about-item">
        <h3 className="about-item-title">Education</h3>
        <div className="about-item-content">
          <p className="about-item-content-date">2015 - 2019</p>
          <div>
            <p className="about-item-content-headline">IT Engineer</p>
            <p className="about-item-content-description">
              Slobomir P University
            </p>
          </div>
        </div>
        <div className="about-item-content">
          <p className="about-item-content-date">2011 - 2015</p>
          <div>
            <p className="about-item-content-headline">Computer Technician</p>
            <p className="about-item-content-description">
              Electrotehnical High School in Doboj
            </p>
          </div>
        </div>
      </div>
    </div>
    <Button variant="btn-outline">Download CV</Button>
  </Section>
);

export default About;
