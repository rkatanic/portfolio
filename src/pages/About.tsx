import Section from "../components/Section";
import Button from "../components/Button";
import { CV_LINK } from "../util/constants/links";

import "./About.css";

const About = (): JSX.Element => {
  const handleCvLinkOpen = () => window.open(CV_LINK, "_blank");

  return (
    <Section id="about">
      <h3 className="headline">My personal journey</h3>
      <div className="about">
        <div className="about-item">
          <h3 className="card-title">Experience</h3>
          <div className="about-item-content">
            <p className="about-item-content-date">08/2019 - present</p>
            <div className="about-item-content-wrapper">
              <p className="about-item-content-headline">Software Developer</p>
              <p className="about-item-content-description">ProductDock</p>
              <p className="about-item-content-location">Doboj</p>
            </div>
          </div>
          <div className="about-item-content">
            <p className="about-item-content-date">02/2019 - 05/2019</p>
            <div className="about-item-content-wrapper">
              <p className="about-item-content-headline">Intern</p>
              <p className="about-item-content-description">
                codecentric bosnia rookie boot camp
              </p>
              <p className="about-item-content-location">Doboj</p>
            </div>
          </div>
        </div>
        <div className="about-item">
          <h3 className="card-title">Education</h3>
          <div className="about-item-content">
            <p className="about-item-content-date">09/2015 - 09/2019</p>
            <div className="about-item-content-wrapper">
              <p className="about-item-content-headline">IT Engineer</p>
              <p className="about-item-content-description">
                Slobomir P University
              </p>
              <p className="about-item-content-location">Doboj</p>
            </div>
          </div>
          <div className="about-item-content">
            <p className="about-item-content-date">09/2011 - 05/2015</p>
            <div className="about-item-content-wrapper">
              <p className="about-item-content-headline">Computer Technician</p>
              <p className="about-item-content-description">
                Electro-tehnical High School
              </p>
              <p className="about-item-content-location">Doboj</p>
            </div>
          </div>
        </div>
      </div>
      <Button variant="btn-outline" onClick={handleCvLinkOpen}>
        Download CV
      </Button>
    </Section>
  );
};

export default About;
