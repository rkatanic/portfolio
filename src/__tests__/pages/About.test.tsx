import { fireEvent, render } from "@testing-library/react";
import About from "../../pages/About";

describe("About", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<About />);

    expect(baseElement).toMatchSnapshot();
  });

  it("should open cv link", (): void => {
    global.open = jest.fn();
    const { getByText } = render(<About />);

    fireEvent.click(getByText("Download CV"));

    expect(global.open).toHaveBeenNthCalledWith(
      1,
      "https://drive.google.com/file/d/1SYQCLxy7JCt7TJfHK5h1G7VqL4mhM8NM/view",
      "_blank"
    );
  });
});
