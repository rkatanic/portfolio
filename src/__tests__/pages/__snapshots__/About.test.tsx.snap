// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render 1`] = `
<body>
  <div>
    <section
      class="section"
      id="about"
    >
      <h3
        class="headline"
      >
        My personal journey
      </h3>
      <div
        class="about"
      >
        <div
          class="about-item"
        >
          <h3
            class="card-title"
          >
            Experience
          </h3>
          <div
            class="about-item-content"
          >
            <p
              class="about-item-content-date"
            >
              08/2019 - present
            </p>
            <div
              class="about-item-content-wrapper"
            >
              <p
                class="about-item-content-headline"
              >
                Software Developer
              </p>
              <p
                class="about-item-content-description"
              >
                ProductDock
              </p>
              <p
                class="about-item-content-location"
              >
                Doboj
              </p>
            </div>
          </div>
          <div
            class="about-item-content"
          >
            <p
              class="about-item-content-date"
            >
              02/2019 - 05/2019
            </p>
            <div
              class="about-item-content-wrapper"
            >
              <p
                class="about-item-content-headline"
              >
                Intern
              </p>
              <p
                class="about-item-content-description"
              >
                codecentric bosnia rookie boot camp
              </p>
              <p
                class="about-item-content-location"
              >
                Doboj
              </p>
            </div>
          </div>
        </div>
        <div
          class="about-item"
        >
          <h3
            class="card-title"
          >
            Education
          </h3>
          <div
            class="about-item-content"
          >
            <p
              class="about-item-content-date"
            >
              09/2015 - 09/2019
            </p>
            <div
              class="about-item-content-wrapper"
            >
              <p
                class="about-item-content-headline"
              >
                IT Engineer
              </p>
              <p
                class="about-item-content-description"
              >
                Slobomir P University
              </p>
              <p
                class="about-item-content-location"
              >
                Doboj
              </p>
            </div>
          </div>
          <div
            class="about-item-content"
          >
            <p
              class="about-item-content-date"
            >
              09/2011 - 05/2015
            </p>
            <div
              class="about-item-content-wrapper"
            >
              <p
                class="about-item-content-headline"
              >
                Computer Technician
              </p>
              <p
                class="about-item-content-description"
              >
                Electro-tehnical High School
              </p>
              <p
                class="about-item-content-location"
              >
                Doboj
              </p>
            </div>
          </div>
        </div>
      </div>
      <button
        class="btn btn-outline"
        type="button"
      >
        Download CV
      </button>
    </section>
  </div>
</body>
`;
